### PR TITLE
PYTHON-3075 [Backport-3.13] bulk_write does not apply CodecOptions to upserted_ids result

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
evergreen patch

**Comment by ShaneHarvey** [comment]
evergreen patch

File: test/test_bulk.py [hunk 1]
```diff
@@ -361,6 +365,79 @@ def test_bulk_write_invalid_arguments(self):
...
+             'upserted': [{'index': 0, '_id': uuids[0]},
+                          {'index': 1, '_id': uuids[1]},
+                          {'index': 2, '_id': uuids[2]}]},
+            result.bulk_api_result)
+
+    def test_upsert_uuid_standard_subdocuments(self):
+        options = CodecOptions(uuid_representation=UuidRepresentation.STANDARD)
+        coll = self.coll.with_options(codec_options=options)
+        ids = [
+            {'f': Binary(bytes(i)), 'f2': uuid.uuid4()}
```
Comment (ShaneHarvey) [inline test/test_bulk.py:414]: This test runs into a bytes/binary issue on python 2. Is there any point of using `'f': Binary(bytes(i))`? What if we just did `'f': 1`?

File: pymongo/pool.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
     HAS_SNI as _HAVE_SNI,
     IPADDR_SAFE as _IPADDR_SAFE)
 
-from bson import DEFAULT_CODEC_OPTIONS
+from bson import DEFAULT_CODEC_OPTIONS, codec_options
```
Comment (ShaneHarvey) [inline pymongo/pool.py:30]: Importing codec_options is not needed. Is your editor adding this automatically by accident?

Comment (blink1073) [reply pymongo/pool.py:30]: Yeah, I just disabled it

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**